### PR TITLE
Use uint64 height types instead of int64

### PR DESCRIPTION
**Pull request by Manav-Aggarwal**
## Overview
Closes: #1176 

## Checklist

- [x] New and updated code has appropriate documentation
- [x] New and updated code has new and/or updated testing
- [x] Required CI checks are passing
- [ ] Visual proof for any user facing features like CLI or documentation updates
- [x] Linked issues closed with keywords

### CODE REVIEW

File: types/test_utils.go [hunk 1]
```diff
@@ -36,7 +36,7 @@ func GetRandomSignedHeader() (*SignedHeader, ed25519.PrivKey, error) {
 
 			BaseHeader: BaseHeader{
 				ChainID: "test",
-				Height:  rand.Uint64(), //nolint:gosec,
+				Height:  uint64(rand.Int63()), //nolint:gosec,
```
Comment (MSevey) [inline types/test_utils.go:39]: I'm confused at what is happening here. Why don't we continue to use `rand.Uint64`?

**Review by MSevey — APPROVED** [review]
lgtm, one minor question.

**Review by tzdybal — APPROVED** [review]
LGTM 💪

File: types/test_utils.go (hunk 1 above)
Comment (Manav-Aggarwal) [reply types/test_utils.go:39]: If we use `rand.Uint64` and get a number bigger than the positive range of int64, converting it to int64 will make it go negative, so better to stick to `rand.Int63()` which `returns a non-negative pseudo-random 63-bit integer as an int64`

**Pull request merged**